Overzealous exception blocking from dodola.service.log_service()

`dodola.service.log_service()` currently decorates all services and logs their entrance and exit. It also logs `Exceptions` but it is a bit too heavy handed. Exceptions raised when the stack enters different services are correctly logged when run from (for e.g.) a container in a cluster but `log_service()` also blocks exceptions raised from automated system tests. In the case of automated system tests, `pytest` or CI/CD usually just gives a stacktrace describing a very confused `open_zarr()` looking for a Zarr store or Group that doesn't exist.

Here is example output where I put a `raise ValueError("uh-oh!")` at the opening of `dodola.services.regrid()` to get:

```
FAILED                           [ 50%]
dodola/tests/test_services.py:154 (test_regrid[bilinear])
regrid_method = 'bilinear'

    @pytest.mark.parametrize("regrid_method", ["bilinear", "conservative"])
    def test_regrid(regrid_method):
        """Test that services.regrid regrids output"""
        goal_shape = (360, 90)
    
        # Make fake input data.
        ds_in = grid_global(30, 20)
        ds_in["fakevariable"] = wave_smooth(ds_in["lon"], ds_in["lat"])
    
        fakestorage = memory_repository(
            {
                "a_file_path": ds_in,
            }
        )
    
        regrid(
            "input_path",
            out="output_ds",
            method=regrid_method,
            storage=fakestorage,
        )
>       actual_shape = fakestorage.read("output_ds")["fakevariable"].shape

test_services.py:176: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../repository.py:51: in read
    x = open_zarr(self.get_mapper(url_or_path))
../../../../miniconda3/envs/dodola_dev/lib/python3.8/site-packages/xarray/backends/zarr.py:675: in open_zarr
    ds = open_dataset(
../../../../miniconda3/envs/dodola_dev/lib/python3.8/site-packages/xarray/backends/api.py:572: in open_dataset
    store = opener(filename_or_obj, **extra_kwargs, **backend_kwargs)
../../../../miniconda3/envs/dodola_dev/lib/python3.8/site-packages/xarray/backends/zarr.py:296: in open_group
    zarr_group = zarr.open_group(store, **open_kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

store = <fsspec.mapping.FSMap object at 0x7fe48a162f40>, mode = 'r'
cache_attrs = True, synchronizer = None, path = '', chunk_store = None
storage_options = None

    def open_group(store=None, mode='a', cache_attrs=True, synchronizer=None, path=None,
                   chunk_store=None, storage_options=None):
        """Open a group using file-mode-like semantics.
    
        Parameters
        ----------
        store : MutableMapping or string, optional
            Store or path to directory in file system or name of zip file.
        mode : {'r', 'r+', 'a', 'w', 'w-'}, optional
            Persistence mode: 'r' means read only (must exist); 'r+' means
            read/write (must exist); 'a' means read/write (create if doesn't
            exist); 'w' means create (overwrite if exists); 'w-' means create
            (fail if exists).
        cache_attrs : bool, optional
            If True (default), user attributes will be cached for attribute read
            operations. If False, user attributes are reloaded from the store prior
            to all attribute read operations.
        synchronizer : object, optional
            Array synchronizer.
        path : string, optional
            Group path within store.
        chunk_store : MutableMapping or string, optional
            Store or path to directory in file system or name of zip file.
        storage_options : dict
            If using an fsspec URL to create the store, these will be passed to
            the backend implementation. Ignored otherwise.
    
        Returns
        -------
        g : zarr.hierarchy.Group
    
        Examples
        --------
        >>> import zarr
        >>> root = zarr.open_group('data/example.zarr', mode='w')
        >>> foo = root.create_group('foo')
        >>> bar = root.create_group('bar')
        >>> root
        <zarr.hierarchy.Group '/'>
        >>> root2 = zarr.open_group('data/example.zarr', mode='a')
        >>> root2
        <zarr.hierarchy.Group '/'>
        >>> root == root2
        True
    
        """
    
        # handle polymorphic store arg
        clobber = mode != "r"
        store = _normalize_store_arg(
            store, clobber=clobber, storage_options=storage_options, mode=mode
        )
        if chunk_store is not None:
            chunk_store = _normalize_store_arg(chunk_store, clobber=clobber,
                                               storage_options=storage_options)
        path = normalize_storage_path(path)
    
        # ensure store is initialized
    
        if mode in ['r', 'r+']:
            if contains_array(store, path=path):
                raise ContainsArrayError(path)
            elif not contains_group(store, path=path):
>               raise GroupNotFoundError(path)
E               zarr.errors.GroupNotFoundError: group not found at path ''

../../../../miniconda3/envs/dodola_dev/lib/python3.8/site-packages/zarr/hierarchy.py:1166: GroupNotFoundError
```

--- a/dodola/services.py
+++ b/dodola/services.py
@@ -12,12 +12,14 @@ logger = logging.getLogger(__name__)
 
 def log_service(func):
     """Decorator for dodola.services to log service start and stop"""
+
     @wraps(func)
     def service_logger(*args, **kwargs):
         servicename = func.__name__
         logger.info(f"Starting {servicename} dodola service")
         func(*args, **kwargs)
         logger.info(f"dodola service {servicename} done")
+
     return service_logger
 
 

--- a/dodola/services.py
+++ b/dodola/services.py
@@ -11,17 +11,13 @@ logger = logging.getLogger(__name__)
 
 
 def log_service(func):
+    """Decorator for dodola.services to log service start and stop"""
     @wraps(func)
     def service_logger(*args, **kwargs):
         servicename = func.__name__
-        try:
-            logger.info(f"Starting {servicename} dodola service")
-            func(*args, **kwargs)
-        except Exception:
-            logger.exception(f"Fatal error in {servicename} dodola service")
-        else:
-            logger.info(f"dodola service {servicename} done")
-
+        logger.info(f"Starting {servicename} dodola service")
+        func(*args, **kwargs)
+        logger.info(f"dodola service {servicename} done")
     return service_logger
 
 

